p-token/src/processor/mod.rs

```diff
@@ -84,6 +84,12 @@ const MAX_FORMATTED_DIGITS: usize = u8::MAX as usize + 2;
 
 /// Checks that the account is owned by the expected program.
 #[inline(always)]
+// JC: the previous implementation of the token program didn't hard-code the
+// expected program id, which made it possible for people to deploy their own
+// token programs.
+// Would it make performance worse to use the `program_id` from the entrypoint?
+// If so, we can stick with this, but if it doesn't add overhead, it'd be best
+// to keep the generic program id.
```

Comment:
It uses more CUs to pass around the `program_id` since it is a "pointer" that goes in the stack of function calls and there are places that we would need to pass it to helpers as well.

Reply:
Ah darn, that's too bad. How much worse is it?

Reply:
I noticed changes between `3` to `6` CUs – it depends on how many calls to the helpers the processor makes.

Reply:
Hm, it doesn't seem necessary at the moment, so we can keep the hard-coded number as is